File: JAVA/2085. Count Common Words With One Occurrence.java
class Solution {
    public int countWords(String[] w1, String[] w2) {
        List<String> a = new ArrayList<>();
        List<String> b = new ArrayList<>();
        Map<String, Integer> x = new HashMap<>();
        Map<String, Integer> f1 = new HashMap<>();
        Map<String, Integer> f2 = new HashMap<>();
        int c = 0;

        for (String word : w1) {
            f1.put(word, f1.getOrDefault(word, 0) + 1);
        }
        for (String word : w1) {
            if (f1.get(word) == 1) {
                a.add(word);
            }
        }
        for (String word : w2) {
            f2.put(word, f2.getOrDefault(word, 0) + 1);
        }
        for (String word : w2) {
            if (f2.get(word) == 1) {
                b.add(word);
            }
        }
        a.addAll(b);
        for (String word : a) {
            x.put(word, x.getOrDefault(word, 0) + 1);
        }
        for (Map.Entry<String, Integer> entry : x.entrySet()) {
            if (entry.getValue() == 2) {
                c++;
            }
        }
        return c;
    }
}

File: C++/2085. Count Common Words With One Occurrence.cpp
class Solution {
public:
    int countWords(vector<string>& w1, vector<string>& w2) {
        vector<string> a, b;
        map<string, int> x;
        int i, c = 0;
        map<string, int> f1,f2;
        for (i = 0; i < w1.size(); i++) {
            f1[w1[i]]++;
        }
        for (i = 0; i < w1.size(); i++) {
            if (f1[w1[i]] == 1) {
                a.push_back(w1[i]);
            }
        }
        for (i = 0; i < w2.size(); i++) {
            f2[w2[i]]++;
        }
        for (i = 0; i < w2.size(); i++) {
            if (f2[w2[i]] == 1) {
                b.push_back(w2[i]);
            }
        }
        for (i = 0; i < b.size(); i++) {
            a.push_back(b[i]);
        }
        for (i = 0; i < a.size(); i++) {
            x[a[i]]++;
        }
        for (auto j = x.begin(); j != x.end(); j++) {
            if (j->second == 2) {  
                c++;
            }
        }
        return c;
    }
};
